pytorch_lightning/accelerators/gpu.py

```diff
@@ -47,9 +47,9 @@ def on_train_start(self) -> None:
         with torch.cuda.device(self.root_device):
             torch.cuda.empty_cache()
 
-    def on_train_end(self) -> None:
+    def teardown(self) -> None:
+        self.lightning_module.device = torch.device("cpu")
         # clean up memory
-        self.model.cpu()
```

Comment:
here, the model is wrapped in DistributedDataParallel
if we call `.cpu()` it will not call .cpu() on the modules and instead run the self.apply on the submodules for the parameters.
The fix we can do is call `self.lightning_module.cpu()`

Reply:
Done !